blocks/math.ts

```diff
@@ -528,44 +516,39 @@ Extensions.register(
 /**
  * Mixin with mutator methods to support alternate output based if the
  * 'math_on_list' block uses the 'MODE' operation.
- * @mixin
- * @augments Block
- * @package
- * @readonly
  */
-const LIST_MODES_MUTATOR_MIXIN = {
+const LIST_MODES_MUTATOR_MIXIN: BlockDefinition = {
   /**
    * Modify this block to have the correct output type.
-   * @param {string} newOp Either 'MODE' or some op than returns a number.
-   * @private
-   * @this {Block}
+   *
+   * @param newOp Either 'MODE' or some op than returns a number.
    */
-  updateType_: function(newOp) {
+  updateType_: function(this: Block, newOp: string) {
     if (newOp === 'MODE') {
-      this.outputConnection.setCheck('Array');
+      this.outputConnection!.setCheck('Array');
```

Comment:
The styleguide doesn't quite forbid non-null assertions but does heavily discourage them. We probably should determine our policy here; my personal inclination in a situation like this is to use ? instead, although that admittedly can result in silent failures.

Reply:
I say we can live with silent failures, and pray that whoever discovers this comment when debugging one in the future will forgive me…

Reply:
Actually, I've changed my mind about this.  I think it best to avoid making any functional changes in this PR, but instead try to preserve existing behaviour exactly (and ideally have `tsc` generate `.js` that is identical to the original `.js` file once comments and whitespace are removed).

I agree that using `?.` would likely be preferable here, but I think such code-cleanup should be done in a separate PR.

Reply:
SGTM